johnsonandjohnson/Bodiless-JS
# fix(migration_tool): "UnhandledPromiseRejectionWarning: Error: Page crashed!" thrown by migration tool

## Background
* some migration tool users experience "UnhandledPromiseRejectionWarning: Error: Page crashed!" error during npm run migrate on their local environments. There are no deterministic steps how to reproduce the problem on any local environment.  The error is thrown by puppeteer. 

## Changes
* upgraded puppeteer used by migration tool
* allow migration-settings.json to omit transformers

## Technical notes
* taking into consideration the error is thrown in puppeteer and migration tool uses an old puppeteer version, the hypothesis was made that the error may be fixed in the latest puppeteer version
* puppeteer upgrade solves the problem for users who experienced this problem
* migration tool does not have puppeteer as a direct dependency.  migration tool depends on headless-chrome-crawler that depends on puppeteer, hereby we have to override npm package nested dependency. In order to do it, headless-chrome-crawler was forked and puppeteer was updated in the fork.

## Test Instructions
Test case 1. remove transformers from migration-settings.json and migrate a site
Expected: the site is migrated. no errors is thrown in the terminal
Test case 2. add a transformer to migration-settings.json and migrate a site
Expected: the site is migrated. the transformers are applied

## Related Issues
* https://github.com/yujiosaka/headless-chrome-crawler/issues/350
* https://github.com/puppeteer/puppeteer/issues/1321
* https://stackoverflow.com/questions/15806152/how-do-i-override-nested-npm-dependency-versions
